src/module.indexer/model/dftx/oracle.e2e.ts

```diff
@@ -366,3 +367,68 @@ describe('invalidate set oracle data', () => {
     }
   })
 })
+
+describe('interval set oracle data', () => {
+  const container = new MasterNodeRegTestContainer()
+  let app: NestFastifyApplication
+  let client: JsonRpcClient
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForWalletCoinbaseMaturity()
+
+    app = await createTestingApp(container)
+    client = new JsonRpcClient(await container.getCachedRpcUrl())
+  })
+
+  afterAll(async () => {
+    await stopTestingApp(container, app)
+  })
+
+  it('should get interval', async () => {
+    const address = await container.getNewAddress()
+    const oracleId = await client.oracle.appointOracle(address, [
+      {
+        token: 'S1',
+        currency: 'USD'
+      }
+    ], {
+      weightage: 1
+    })
+    await container.generate(1)
+
+    const oneMinute = 60
+    let mockTime = Math.floor(new Date().getTime() / 1000)
+    for (let h = 0; h < 24; h++) {
+      for (let z = 0; z < 4; z++) {
+        mockTime += (z * 15) * oneMinute
+        await client.misc.setMockTime(mockTime)
+        await container.generate(1)
```

Comment:
```suggestion
        await container.generate(11)
```
Maybe we should wait for 11 blocks. For the median time to change.

Reply:
1 Block:
```
 medianTime: 1647336701,
time: 1647337602
```
11 Blocks:
```
medianTime: 1647338104,
time: 1647338104
```

Reply:
However, waiting for 11 blocks does add into the test time.

Reply:
For regtest 2 blocks is enough.
```
await container.generate(2)
```